com.unity.ml-agents/Tests/Editor/SideChannelTests.cs

```diff
@@ -103,6 +108,58 @@ public void TestFloatPropertiesSideChannel()
             fakeData = RpcCommunicator.GetSideChannelMessage(dictSender);
             RpcCommunicator.ProcessSideChannelData(dictReceiver, fakeData);
             Assert.AreEqual(wasCalled, 1);
+
+            var keysA = propA.ListProperties();
+            Assert.AreEqual(2, keysA.Count);
+            Assert.IsTrue(keysA.Contains(k1));
+            Assert.IsTrue(keysA.Contains(k2));
+
+            var keysB = propA.ListProperties();
+            Assert.AreEqual(2, keysB.Count);
+            Assert.IsTrue(keysB.Contains(k1));
+            Assert.IsTrue(keysB.Contains(k2));
+        }
+
+        [Test]
+        public void TestOutgoingMessageRawBytes()
```

Comment:
There should be the other case for this, where you write a large amount of data, and then set a small amount of data, and ensure that the byte arrays are equal.

Reply:
This writes 8 bytes (int+float), and then does SetRawBytes with 4. How much of a difference do you think is necessary? I confirmed that without `MemoryStream.SetLength(0)` this fails.

Reply:
I can add some comments to make this clearer

Reply:
ah, sorry.  i misread the `byte[]` part for an `int` array.